### ISSUE TITLE
Bizarre crash when using `node_cdr` on numeric or boolean data

### ISSUE THREAD

**Issue by aaronrudkin** [tidyverse/rlang#394]
My development computer: RStudio (R 3.4.2), MacOS 10.13, rlang 0.1.6 (CRAN version). 

I opened an old install of RStudio (R 3.3.1), Windows 10, rlang 0.1.6 (CRAN version), to confirm it happens there as well.

This code causes a reproducible hard crash of the R session:

```
library(rlang)
node_cdr(123) # crashes
node_cdr(NA) # also crashes
node_cdr(TRUE) # also crashes
```

Using other non-language types does not cause the crash:
```
library(rlang)
node_cdr(data.frame(X=1:10))
#  [1]  1  2  3  4  5  6  7  8  9 10
node_cdr("hello world")
# <CHARSXP: "hello world">
node_cdr(NULL)
# NULL
```

**Comment by vspinu** [comment]
This is kinda documented :) You are not supposed to use these functions on anything else than a node (aka linked list). That it does accept a non-node an crashes is bizarre indeed, but well,  #388 was closed today ;)

**Comment by aaronrudkin** [comment]
Ah, I had searched open issues not closed issues. Maintainers can feel free to close if the answer to this is "known, documented, intended to be internal only".

The specific route that led me to this was attempting to extract the body of an arbitrary function call in an arbitrary namespace into a list of language statements.

**Comment by lionel-** [comment]
I didn't know about the `internal` keyword back then. They're now internal so won't show up in the help index.

Bottom line, don't use node accessors and pokers unless you're feeling like an adventurous C programmer :)

**Comment by lionel-** [comment]
I have improved the documentation to make it a bit scarier : https://github.com/tidyverse/rlang/commit/73d8f501cbb03eba41206839e77077fdce1446c2

@aaronrudkin may I ask how you found these functions?

**Comment by aaronrudkin** [comment]
@lionel- 

They were in the website docs, but also I was examining `lang_head` and `lang_tail`, recognized that they were basically type-checking wrappers for `node_car` and `node_cdr`. I checked `node_car` and `node_cdr`'s source internally, saw they were `.Call` wrappers, hoped on the website, found the docs, said "hmmm, I wonder what these do when hit with different kind of data"

In particular, the motivating example I was fiddling with was looking at how closures performed e.g.

```
library(rlang)
mwe <- function(x) { x + 1 + 2 + 3 }
lang_head(mwe) # Error: is_lang(lang) is not TRUE
typeof(mwe) # Closure, as you'd expect
node_car(mwe) # Works -- no crash, interprets function closure as the function
node_cdr(mwe) # Works -- no crash, interprets function closure as the function
```

And when I saw that happened, I figured I'd just hit the function with all the different types. I'd never have reported an issue if the errors were internal, but it's rare you see a function actually crash the R session so I wasn't sure if anyone was aware.
